services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestHandler.java

```diff
@@ -313,8 +314,8 @@ public void channelRead(ChannelHandlerContext context, Object message) throws Ex
           }
           context.writeAndFlush(response);
         } catch (VeniceNoStoreException e) {
-          context.writeAndFlush(
-              new HttpShortcutResponse("No storage exists for: " + e.getStoreName(), HttpResponseStatus.BAD_REQUEST));
+          HttpResponseStatus status = getHttpResponseStatus(e);
```

Comment:
I got a question here: When it throw NoStoreException, does it mean storeRepository cannot find the store? If that's the case, the new method you add should not help right?

Reply:
No, it throws that even if cannot see the storage engine loaded, which is kind of misnomer. we should ideally throw a new NoStorageEngineFoundException in getStorageEngineOrThrow.

Reply:
Oh I see. So this is the II CV bug fix on router side then.

Reply:
right